openedx/core/djangoapps/user_api/accounts/tests/pact/PactBypassAuthMiddleware.py

```diff
@@ -0,0 +1,36 @@
+"""
+Contain the middleware logic needed during pact verification
+"""
+from django.contrib import auth
+from django.utils.deprecation import MiddlewareMixin
+from django.conf import settings
+from django.core.exceptions import MiddlewareNotUsed
+
+User = auth.get_user_model()
+
+
+class AuthenticationMiddleware(MiddlewareMixin):
+    """
+    Middleware to add default authentication into the requests for pact verification.
+
+    This middleware is required to add a default authenticated user and bypass CSRF validation
+    into the requests during the pact verification workflow. Without the authentication, the pact verification
+    process will not work as the apis.
+    See https://docs.pact.io/faq#how-do-i-test-oauth-or-other-security-headers
+    """
+
+    def __init__(self, get_response):
+        if getattr(settings, 'PACT_BYPASS_AUTHENTICATION', None) is False:
+            raise MiddlewareNotUsed
+        super().__init__(get_response)
```

Comment:
I'm not sure if there's any harm in having the superclass constructor call here, but I've only ever seen it as the very first statement before. Might be good to move it to above the `if` block.